src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +892,49 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                throw new InternalExpressionException("Cannot change the gamemode of non-player entities");
+            }
+            
+            switch(v.getString().toLowerCase()){
+                case "survival":((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);return;
+                case "creative":((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);return;
+                case "adventure":((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);return;
+                case "spectator":((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);return;
+            }
+
+            switch((int)NumericValue.asNumber(v).getLong()){
```

Comment:
Check value type first and then use numbers, otherwise check string.

Reply:
idk how to do

Reply:
check instanceof of the 'v'

Reply:
ah tru

Reply:
wai,t it wont let me do if(v instanceof String)
It's an instance of a carpet.script.value.Value